### DISCUSSION TITLE
Routes in app/api is not working

### DISCUSSION THREAD

**Discussion by lmai5** [vercel/next.js#61362] [App Router] [answered]
As the title states, the routes inside the `app/api` directory in my project is somehow interpreted as a page (not found page returned). Routes in other directories are working fine. Anyone ever encountered this kind of issue? 
Note that, my project is upgraded from Page Router to App Router

**Comment by icyJoseph** [comment]
Could you share your file structure?

**Reply by ik04** [reply]
saved me alot of pain, thank you

**Comment by lmai5** [comment]
Thanks for replying Joseph. Here's my project structure. So basically routes in the `app\api` is not being interpreted as APIs.  
<img width="226" alt="image" src="https://github.com/vercel/next.js/assets/6018147/b041730b-57ab-4b4c-bc1a-7857048216ed">

I've updated all NextJS related frameworks to the latest one.

**Comment by lmai5** [comment, marked as answer]
Dang it, I have figured out the cause. There's a space at the end of the api folder name. Wasted me 2 days.

**Comment by shvamabps** [comment]
I am facing same issue but I have checked there are no spaces at the end or at the start of the directory name.
